kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProduceRateLimiters.java

```diff
@@ -35,14 +37,21 @@
   private final boolean rateLimitingEnabled;
   private final LoadingCache<String, RequestRateLimiter> countCache;
   private final LoadingCache<String, RequestRateLimiter> bytesCache;
+  private final RequestRateLimiter bytesLimiterGlobal;
+  private final RequestRateLimiter countLimiterGlobal;
 
   @Inject
   public ProduceRateLimiters(
       @ProduceRateLimiterCount Provider<RequestRateLimiter> countLimiterProvider,
       @ProduceRateLimiterBytes Provider<RequestRateLimiter> bytesLimiterProvider,
+      @ProduceRateLimiterCountGlobal RequestRateLimiter countLimiterGlobal,
+      @ProduceRateLimiterBytesGlobal RequestRateLimiter bytesLimiterGlobal,
       @ProduceRateLimitEnabledConfig Boolean produceRateLimitEnabledConfig,
       @ProduceRateLimitCacheExpiryConfig Duration produceRateLimitCacheExpiryConfig) {
     this.rateLimitingEnabled = requireNonNull(produceRateLimitEnabledConfig);
+    this.countLimiterGlobal = requireNonNull(countLimiterGlobal);
+    this.bytesLimiterGlobal = requireNonNull(bytesLimiterGlobal);
```

Comment:
Not something addressable as part of this change, but I'm starting to wonder - does it really make sense to use HK2 for the actual rate-limiter instance management (as opposed to the rate-limiter config management)? It makes setting up the testing mocks a bit easier, but that can be worked around (e.g. with a delegating ctor), and the price for it is a significant amount of factories and annotations.
Thoughts, @ehumber , @rigelbm ?

Reply:
I'm not keen on the HK2 usage here, in my view it makes thing more complicated than they need to be.  We also know that H2K adds performance costs (although I don't know how much of an impact that will have for this particular code)

The testing isn't actually very good with the injection - the hard bit of getting this change working was getting the injection working, and you can't mock that, you need to run an actual rest instance.